foreshadow/preprocessor.py

```diff
@@ -338,19 +338,20 @@ def serialize(self):
 
         """
         json_cols = {
-            k: (
-                self._intent_map[k].__name__,
-                serialize_pipeline(
+            k: {
+                "intent": self._intent_map[k].__name__,
+                "pipeline": serialize_pipeline(
                     self._pipeline_map.get(k, Pipeline([("null", None)]))
                 ),
-                [c[1].__name__ for c in self._choice_map[k]],
-            )
+                "all_matched_intents": [c[1].__name__ for c in self._choice_map[k]],
+            }
             for k in self._intent_map.keys()
         }
 
         # Serialize multi-column processors
         json_multi = [
-            [v[0], v[1], serialize_pipeline(v[2])] for v in self._multi_column_map
+            {"name": v[0], "columns": v[1], "pipeline": serialize_pipeline(v[2])}
```

Comment:
Why is this part still using list format?

Reply:
Internally it is still a list. Only the JSON configuration was changed, it mapped to the same thing internally (which uses a list).